plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQuerySplitSource.java

```diff
@@ -135,22 +135,21 @@ private List<BigQuerySplit> getSplits(
         TupleDomain<ColumnHandle> tableConstraint = bigQueryTableHandle.constraint();
         Optional<String> filter = BigQueryFilterQueryBuilder.buildFilter(tableConstraint);
 
+        TableId remoteTableId;
+        TableDefinition.Type tableType;
+        boolean useStorageApi;
+
         if (bigQueryTableHandle.isQueryRelation()) {
             BigQueryQueryRelationHandle bigQueryQueryRelationHandle = bigQueryTableHandle.getRequiredQueryRelation();
             List<BigQueryColumnHandle> columns = bigQueryTableHandle.projectedColumns().orElse(ImmutableList.of());
-            boolean useStorageApi = bigQueryQueryRelationHandle.isUseStorageApi();
+            useStorageApi = bigQueryQueryRelationHandle.isUseStorageApi();
 
             // projectedColumnsNames can not be used for generating select sql because the query fails if it does not
             // include a column name. eg: query => 'SELECT 1'
             String query = filter
                     .map(whereClause -> "SELECT * FROM (" + bigQueryQueryRelationHandle.getQuery() + " ) WHERE " + whereClause)
                     .orElseGet(bigQueryQueryRelationHandle::getQuery);
 
-            if (emptyProjectionIsRequired(bigQueryTableHandle.projectedColumns())) {
-                String sql = "SELECT COUNT(*) FROM (" + query + ")";
-                return createEmptyProjection(session, sql);
-            }
-
             if (!useStorageApi) {
                 log.debug("Using Rest API for running query: %s", query);
                 return List.of(BigQuerySplit.forViewStream(columns, filter));
```

Comment:
For REST API with empty projection ? Now we won't be handling them right ? Or would that situation won't occur today

Reply:
the empty projection for REST API will be handled by this code.

```
        return emptyProjectionIsRequired(bigQueryTableHandle.projectedColumns())
                ? createEmptyProjection(session, tableType, remoteTableId, filter, limit)
                : readFromBigQuery(session, tableType, remoteTableId, bigQueryTableHandle.projectedColumns(), tableConstraint, limit, useStorageApi);
```